client/packages/inventory/src/Stocktake/api/hooks/line/useStocktakeLines.ts

```diff
@@ -2,11 +2,23 @@ import { useQuery, useUrlQueryParams } from '@openmsupply-client/common';
 import { useStocktakeApi } from '../utils/useStocktakeApi';
 import { useStocktakeNumber } from '../document/useStocktake';
 
-export const useStocktakeLines = (id: string) => {
+export const useStocktakeLines = (id: string, itemCode?: string) => {
   const { queryParams } = useUrlQueryParams({
     initialSort: { key: 'itemName', dir: 'asc' },
     filters: [{ key: 'itemCodeOrName' }],
   });
+
+  if (itemCode) {
```

Comment:
The existance of itemCode changes the behaviour of this hook quite a bit.
Maybe should be a seperate hook? or we could look at one of the other solutions I mention in the issue.